api/server.go

```diff
@@ -0,0 +1,59 @@
+// Package server provides a server object that represents the GoCert backend
+package server
+
+import (
+	"crypto/tls"
+	"net/http"
+	"time"
+
+	"github.com/canonical/gocert/internal/certificates"
+)
+
+// loadServerCertificates takes in a certificate and a private key and determines
+// whether to use self signed or given certificates, then returns it in a format
+// expected by the server
+func loadServerCertificates(certificate, key string) (*tls.Certificate, error) {
+	if certificate == "" || key == "" {
+		caCertPEM, caPrivateKeyPEM, err := certificates.GenerateCACertificate()
+		if err != nil {
+			return nil, err
+		}
+		certificate, key, err = certificates.GenerateSelfSignedCertificate(caCertPEM, caPrivateKeyPEM)
+		if err != nil {
+			return nil, err
+		}
+	}
+	serverCerts, err := tls.X509KeyPair([]byte(certificate), []byte(key))
+	if err != nil {
+		return nil, err
+	}
+	return &serverCerts, nil
+}
+
+// NewServer creates a new http server with handlers that Go can start listening to
+func NewServer(certificate, key string) (*http.Server, error) {
+	serverCerts, err := loadServerCertificates(certificate, key)
+	if err != nil {
+		return nil, err
+	}
+	m := http.NewServeMux()
+	m.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		_, err := w.Write([]byte("Hello World"))
+		if err != nil {
+			return
```

Comment:
Shouldn't this return nil and the error?

Reply:
No, the http handler interface requires returning nothing.

Reply:
Oh, my bad, I did not realize this was the return for the anonymous function.